### ISSUE TITLE
Names?

### ISSUE THREAD

**Issue by adamlauretig** [zamorarr/tensorr#7]
Thank you very much for this package, it's very cool! I was wondering if we could get the ability to add names for each dimension of the tensor? Something like a generalization of rownames?

**Comment by zamorarr** [comment]
Hey! Thanks for using the package. I think this can be done for sparse tensors, but just to make sure can you give an example of how you'd like to use this functionality? Would you like to be able to index via the dimnames instead of the subscripts/indices?

Here's an example using a base R multi-dimensional array. Run the code in R and let me know if this is along the line of what you are proposing.  Thanks!

```R
mytensor <- array(1:12, dim = c(2,2,3))
mytensor <- provideDimnames(mytensor, base = list(LETTERS, letters, month.abb))
mytensor[,,"Feb"]
```

**Comment by adamlauretig** [comment]
Yep, that's what I had in mind! I'm working w/labeled data (word counts), and the ability to access an entry like: spwordtensor["cat", "dog", "chase"]  (returning a 1x1x1 cell), or spwordtensor["cat", , ] (returning a sparse matrix) would be great.

Thanks again!

**Comment by zamorarr** [comment]
Closed in #9
